evap/contributor/tests/test_views.py

```diff
@@ -272,6 +274,26 @@ def test_display_request_buttons(self):
         self.assertEqual(page.body.decode().count("Request changes"), 0)
         self.assertEqual(page.body.decode().count("Request creation of new account"), 2)
 
+    def test_contributor_evaluation_edit_main_language(self):
+        self.evaluation.main_language = "en"
+        self.evaluation.save()
+
+        response = self.app.get(self.url, user=self.editor, status=200)
+        form = response.forms["evaluation-form"]
+
+        form["main_language"] = Evaluation.UNDECIDED_MAIN_LANGUAGE
+        page = form.submit(name="operation", value="save")
+        self.assertContains(page, "A decision on the main language has to be made.")
+
+        for lang, __ in settings.LANGUAGES:
+            form["main_language"] = lang
+            form.submit(name="operation", value="save")
+            self.assertEqual(Evaluation.objects.get().main_language, lang)
+
+        with self.assertRaises(ValueError):
+            form["main_language"] = "some_other_wrong_value"
+            form.submit(name="operation", value="save")
```

Comment:
The raise is already on the assignment before
```suggestion
```

This raises (ha) the question whether we want to have a test that actually carries out the request; I think it's fine as is, the only pro argument for me would be coverage

Reply:
still an issue with the current code :D